representative_mapping.py

```diff
@@ -0,0 +1,38 @@
+from json import loads
+from threading import Lock
+from time import sleep
+
+from requests import get
+
+
+class representative_mapping:
+    def __init__(self):
+        self.__mutex = Lock()
+        self.__representative_mappings: list[dict] = []
+
+    def load_from_file(self, path: str):
+        try:
+            with open(path) as file:
+                with self.__mutex:
+                    self.__representative_mappings = loads(file.read())
+        except FileNotFoundError:
+            pass
+
+    def load_from_url(self, url: str):
+        with self.__mutex:
+            self.__representative_mappings = get(url).json()
+
+    def load_from_url_loop(self, url: str, delay_seconds: int):
+        while True:
+            try:
+                self.load_from_url(url)
+            finally:
+                sleep(delay_seconds)
+
+    def find(self, node_id: str, address: str) -> dict:
```

Comment:
Let's do an exhaustive search and look for any extra matches.
And return a list of matches and then log the duplicates as warnings.

Reply:
The reason for doing is that we are open to man-in-the-middle attacks (MITM) and MITM attacks are actively performed on a regular basis. So we want to log them to see the extent of them.

Reply:
So if there are multiple matches, which one should be displayed in the website?

Reply:
Let's start by displaying them all. But what is important is that we establish the concept of warnings channel/stream where warning messages can go. On the website, the warnings could be displayed in a log panel/section.

Reply:
Let's do that as a separate PR.